test/erlfmt_SUITE.erl

```diff
@@ -99,6 +99,13 @@ end_per_group(_GroupName, _Config) ->
     ok.
 
 init_per_testcase(_TestCase, Config) ->
+    GroupProps = ?config(tc_group_properties, Config),
+    case proplists:get_bool(syntax_tools, GroupProps) of
+        true ->
+            put('$syntax_tools$', true);
```

Comment:
Wow I didn't know about this magic.
I know this is a minimal diff, but thinking of the next person reading this, maybe it would be better to use a parameter, since each test already has access to Config.  Something like:

```
is_syntax_tools(Config) ->
    proplists:get_bool(syntax_tools, ?config(tc_group_properties, Config)).
    
 some_test(Config) ->
   RoundTripWithSyntaxTools = is_syntax_tools(Config)
   ...
   parse_form(Form, RoundTripWithSyntaxTools)
```

But I think this was a great way to get test coverage and it is really impressive that everything is passing.

Reply:
I like to pass things as parameters whenever I can, but all the test code would have to get the extra Options parameter passed all the way down to the parse_form/parse_forms functions, and that seemed too intrusive. I also thought about using meck to modify the behaviour of the test code, but that's also a bit yucky and causes a dep on meck. What do you prefer?

Reply:
I think I prefer a parameter, since I don't think it needs to pass too far down.
But let's also hear what @michalmuskala has to say.

Reply:
I think this is fine. It's not pretty, but it's practical. I actually wonder if we should try doing something in `erlfmt_format_SUITE` as well to leverage all the examples we have in there, though I'm not 100% sure how that would look like.